code/modules/holodeck/computer.dm

```diff
@@ -153,6 +153,8 @@ and clear when youre done! if you dont i will use :newspaper2: on you
 			var/list/checked = program_cache.Copy()
 			if (obj_flags & EMAGGED)
 				checked |= emag_programs
+				log_game("[key_name(usr)] loaded a restricted Holodeck program: [program_to_load].")
```

Comment:
this is checking if the holodeck loads a program while emagged, not that its loading a program thats in the emag_programs list (which are all restricted).

Reply:
Ah, right. How would you go about it?